src/components/App/App.tsx

```diff
@@ -1,98 +1,88 @@
 import './App.css';
 import { CleanArticle, fetchNewsData, getSentiment } from '../../apiCalls'
 
-interface AppState {
-  articles: CleanArticle[],
-  userSentiment: number | string
-}
 
-interface AppProps {}
+const App = (): JSX.Element => {
 
-class App extends React.Component<AppProps, AppState> {
-  constructor(props: AppProps) {
-    super(props)
-    this.state = {
-      articles: [],
-      userSentiment: ''
-    }
-  }
+  const [ articles, setArticles ] = useState<CleanArticle[]>([])
+  const [ userSentiment, setUserSentiment ] = useState<number | null>(null)
+  const [ error, setError ] = useState('')
 
-  componentDidMount = (): void => {
+  useEffect((): void => {
     fetchNewsData()
-      .then((cleanArticles): any => {
-        this.getSentimentScores(cleanArticles)
-          .then(response => {
+      .then((cleanArticles: CleanArticle[]): void => {
+        getSentimentScores(cleanArticles)
+          .then((response: number[]) => {
 
             const scoredArticles = cleanArticles.map((article, i) => {
                article.sentiment = response[i] || 0
                return article;
             })
 
-            this.setState({ articles: scoredArticles })
+            setArticles(scoredArticles)
           })
       })
-  }
+      .catch(error => setError(error.message))
+  }, [])
 
-  getSentimentScores = (cleanArticles: CleanArticle[]): Promise<any> => {
-    return (Promise as any).all(
-      cleanArticles.map((article: CleanArticle, i: number) => {
+  const getSentimentScores = (cleanArticles: CleanArticle[]): Promise<number[]> => {
+    return Promise.all(
+      cleanArticles.map((article: CleanArticle) => {
         return getSentiment(article.abstract)
       })
     )
   }
 
-  changeUserSentiment = (newSentiment: number) => {
-    this.setState({ userSentiment: newSentiment })
-    this.sortBySentiment(newSentiment)
+  useEffect((): void => {
+    sortBySentiment(userSentiment)
+  }, [ userSentiment ])
+
+  const changeUserSentiment = (newSentiment: number) => {
+    setUserSentiment(newSentiment)
   }
 
-  sortBySentiment = (newSentiment: number): void => {
-    const { articles } = this.state;
-    const articlesCopy = articles.slice();
-    let sortedArticles: CleanArticle[];
+  const sortBySentiment = (newSentiment: number|null): void => {
+    let sortedArticles : CleanArticle[] = [];
 
     if (newSentiment === -1) {
-      sortedArticles = articlesCopy.sort((articleA, articleB) => {
+      sortedArticles = articles.sort((articleA, articleB) => {
         return articleB.sentiment - articleA.sentiment;
       })
-
-      this.setState({ articles: sortedArticles })
-
     } else if (newSentiment === 1) {
-      sortedArticles = articlesCopy.sort((articleA, articleB) => {
+      sortedArticles = articles.sort((articleA, articleB) => {
         return articleA.sentiment - articleB.sentiment;
       })
-
-      this.setState({ articles: sortedArticles })
+    } else {
```

Comment:
We can use this way to shuffle them for now. I would like to try @najinl's approach of having different arrays for sorting them by positive, negative, neutral, and so on.